### ISSUE TITLE
Useful Links to Color Lists

### ISSUE THREAD

**Issue by tajmone** [meodai/color-names#50]
Since I've searched for color-names lists in the past for my [Name That Color] project, I'm pasting here some links from my findings, hoping they might be useful.

This Issue could also be kept alive to allow more links to be contributed below.

## The Color Thesaurus — by Ingrid Sundberg

- http://ingridsundberg.com/2014/02/04/the-color-thesaurus/ (dead link)
- https://web.archive.org/web/20150905154524/http://ingridsundberg.com/2014/02/04/the-color-thesaurus/

Unfortunately, the original page is no longer reachable due to the account being suspended, so I've provided a link via WaybackMachine.

When Ingrid Sundberg's published her Color Thesaurus it had quite a resonance, and it inspired various projects on GitHub. I'm linking a couple of them below. 

### Color-Thesaurus JavaScript

- https://github.com/Tomegreene/Color-Thesaurus

> This is intended to be an extension of the color thesaurus from: http://ingridsnotes.wordpress.com/2014/02/04/the-color-thesaurus/
> 
> I am aiming to collapse things down to a single viewable grid with a drop down list for all the different categories of colors. Also it will include the hex values along with English names for the colors to help developers, and designers communicate clearly.
> 
> It is using Angular javascript framework which might be a bit overkill.

### The New Defaults: A Sass Color Thesaurus

- https://github.com/dudleystorey/thenewdefaults

> A Sass replacement for the standard CSS named color system. Provides more visually appealing hues for many CSS keywords; adds new colors with more relevant and easily remembered names. Inspired by __Ingrid Sundberg__’s color thesaurus and the work of __Adam Morse__.

## Multi-lingual Color Thesaurus

(unrelated to Ingrid Sundberg's Thesaurus)

- https://ivrl.epfl.ch/research-2/research-current/research-image-mining-html/colorthesaurus-2/

License: CC-BY-NC-SA

> We present a color thesaurus with over 9000 color names in ten different languages. Instead of using conventional psychophysical experiments, we use a statistical framework that is based on search results from Google Image Search...

This is an academic project, well documented and with software tools. Beware of the license terms though, as they might be incompatible with other licenses.

## Neural-Network Generated Color Names

- [New paint colors invented by neural network](http://aiweirdness.com/post/160776374467/new-paint-colors-invented-by-neural-network) (original Tumblr post by Janelle Shane)
- [This Is What Happened When an Algorithm Invented New Colors](https://www.nbcnews.com/mach/technology/what-happened-when-algorithm-invented-new-colors-n762121) (aticle reviewing the research)

> nique new colors [...] generated by a neural algorithm trained by Janelle Shane, an optics research scientist, who taught it to analyze patterns in colors and dream up new hues in the style of Sherwin-Williams.

I've found the list of color names artificially produced by IA quite interesting.

[Name That Color]: /github.com/tajmone/name-that-color

**Comment by meodai** [comment]
Wow @tajmone thanks for this! 
- I think I had added tthe colors for "the color thesaurus" manually, I will check if I actually did or only planed to ;) 
- I had seen the colors from http://aiweirdness.com/post/160776374467/new-paint-colors-invented-by-neural-network, and I also found it super interesting, but too many of the names are actually not that good, I think it could be interesting to it with our list to check out, if the outcome would be better. I did not dig into machine learning at all for now though

**Comment by tajmone** [comment]
> I had seen the colors from http://aiweirdness.com/post/160776374467/new-paint-colors-invented-by-neural-network, and I also found it super interesting, but too many of the names are actually not that good,

Yes, I had also come to the same conclusion, but I was quite impressed by the whole approach and many colors did proof to be interesting. Hard to know if IA could contribute in this direction, so right now we might take it with a pinch of salt and consider it just a curiosity — but it's definitely an experiment worth mentioning in any colors-naming project.

> I think it could be interesting to it with our list to check out, if the outcome would be better. I did not dig into machine learning at all for now though.

It might be doable, and there are some good neural network frameworks and libraries in the Node/JS echo system that could be used:

- https://brain.js.org/ (this JS demo deals with color contrast reckognition)
- http://caza.la/synaptic/#/

JS tutorials on the topic are plenty:

- [Neural networks in JavaScript](https://scrimba.com/g/gneuralnetworks)
- [How to create a Neural Network in JavaScript in only 30 lines of code](https://medium.freecodecamp.org/how-to-create-a-neural-network-in-javascript-in-only-30-lines-of-code-343dafc50d49)

Setting up a simple perceptron NNA should be straightfoward, and it might be used to develop an alogrithm to learn from the manual decision on how duplicate color entries [are being/should be] handled. Here's an implementation in 140 lines:

- https://gist.github.com/primaryobjects/dfb8927f9f0ca21b6a24647168cead41

There are enough entries in the colors lists to provide and adequate amount of starting data, the trickiest part would be to establish the criteria by which the machine should classify the problem and learn its lessons.

Could become an "experimental" feature on the side.

**Comment by meodai** [comment]
Would love to dive into that! Definitely let us know here, when you got some time!
